modules/util/source-formatter/src/main/java/com/liferay/source/formatter/check/UpgradeJavaCaptchaUtilCheck.java

```diff
@@ -0,0 +1,72 @@
+/**
+ * SPDX-FileCopyrightText: (c) 2023 Liferay, Inc. https://liferay.com
+ * SPDX-License-Identifier: LGPL-2.1-or-later OR LicenseRef-Liferay-DXP-EULA-2.0.0-2023-06
+ */
+
+package com.liferay.source.formatter.check;
+
+import com.liferay.petra.string.StringBundler;
+import com.liferay.petra.string.StringPool;
+import com.liferay.petra.string.StringUtil;
+import com.liferay.portal.kernel.util.ArrayUtil;
+import com.liferay.source.formatter.check.util.JavaSourceUtil;
+
+import java.util.List;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+/**
+ * @author Renato Rêgo
+ */
+public class UpgradeJavaCaptchaUtilCheck
+	extends BaseUpgradeMatcherReplacementCheck {
+
+	@Override
+	protected String formatMatcherIteration(
+		String content, String newContent, Matcher matcher) {
+
+		List<String> parameterList = JavaSourceUtil.getParameterList(
+			content.substring(matcher.start()));
+
+		if (parameterList.size() != 2) {
+			return newContent;
+		}
+
+		String[] parameterTypes = {"ResourceRequest", "ResourceResponse"};
```

Comment:
@natocesarrego, if we move the `parameterTypes` declaration to before the if, we can use the length and not a literal number, what do you think?

```
	List<String> parameterList = JavaSourceUtil.getParameterList(
		content.substring(matcher.start()));

	String[] parameterTypes = {"ResourceRequest", "ResourceResponse"};

	if (parameterList.size()  != parameterTypes.length) {
		return newContent;
	}
```

Reply:
as we agreed offline, I'll keep it the way it is. :slightly_smiling_face: 

Thanks